clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -1380,6 +1381,14 @@ public void close() {
             connectionsByCipher.close();
             connectionsByClient.close();
         }
+
+        public Metrics metrics() {
```

Comment:
Instead of exposing these 2 fields, I considered having a `wrapInstance()` method but since it's pretty hard to test this class, it's not exposed by Selector and it's "hard" to create a Selector instance, so I preferred just adding getters.

There are other network plugins that need updating with KIP-877. So maybe this will change depending how it goes.